### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by sovchenko**
added Company and Department entities

### CODE REVIEW

File: src/main/java/SalaryTaskDemo.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import employees.EmployeeWithSalaryPerHour;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class SalaryTaskDemo {
+    public static void main(String[] args) {
+        List<Company> companies = new ArrayList<>();
+        Collections.addAll(companies,
```
Comment (toleksyn) [inline src/main/java/SalaryTaskDemo.java]: companies.add()

File: src/main/java/SalaryTaskDemo.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                new Company("Microsoft", 5),
+                new Company("Google", 1));
+
+        List<Department> departments = new ArrayList<>();
+        Collections.addAll(departments,
```
Comment (toleksyn) [inline src/main/java/SalaryTaskDemo.java]: same

File: src/main/java/SalaryTaskDemo.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+                new SalesDepartment(),
+                new MarketingDepartment());
+
+        List<Employee> employees = new ArrayList<>();
+        Collections.addAll(employees,
```
Comment (toleksyn) [inline src/main/java/SalaryTaskDemo.java]: same

File: src/main/java/departmentsAndCompany/Company.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package departmentsAndCompany;
+
+import java.util.List;
+
+public class Company {
+    private String name;
+    private int id;
+    //private List<Department> departments;
```
Comment (toleksyn) [inline src/main/java/departmentsAndCompany/Company.java]: add comment or remove

File: src/main/java/departmentsAndCompany/Company.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+    public String getName() {
+        return name;
+    }
+
+    public int getId() {
+        return id;
+    }
+
+
```
Comment (toleksyn) [inline src/main/java/departmentsAndCompany/Company.java]: remove

File: src/main/java/departmentsAndCompany/Department.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
+package departmentsAndCompany;
+
+public class Department {
+    protected String name;
+    protected int id;
+
+
```
Comment (toleksyn) [inline src/main/java/departmentsAndCompany/Department.java]: remove

File: src/main/java/departmentsAndCompany/Company.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
+package departmentsAndCompany;
```
Comment (toleksyn) [inline src/main/java/departmentsAndCompany/Company.java]: unique, split into two

File: src/main/java/departmentsAndCompany/Department.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    public Department(String name, int id){
+        this.name = name;
+        this.id = id;
+    }
+
+
+   */
+
+    public Department() {
```
Comment (toleksyn) [inline src/main/java/departmentsAndCompany/Department.java]: remove

File: src/main/java/departmentsAndCompany/Department.java [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+    }
+
```
Comment (toleksyn) [inline src/main/java/departmentsAndCompany/Department.java]: remove

File: src/main/java/departmentsAndCompany/MarketingDepartment.java [hunk 10]
```diff
@@ -0,0 +1,10 @@
+package departmentsAndCompany;
+
+public class MarketingDepartment extends Department {
+    // some code unique for the  Marketing Department
```
Comment (toleksyn) [inline src/main/java/departmentsAndCompany/MarketingDepartment.java]: remove

File: src/main/java/departmentsAndCompany/SalesDepartment.java [hunk 11]
```diff
@@ -0,0 +1,9 @@
+package departmentsAndCompany;
+
+public class SalesDepartment extends Department {
+    //some unique code fot the SalesDepartment
```
Comment (toleksyn) [inline src/main/java/departmentsAndCompany/SalesDepartment.java]: same

File: src/main/java/employees/Employee.java [hunk 12]
```diff
@@ -0,0 +1,49 @@
+package employees;
+
+import departmentsAndCompany.Company;
+import departmentsAndCompany.Department;
+
+public class Employee {
```
Comment (toleksyn) [inline src/main/java/employees/Employee.java]: add contract

Comment (toleksyn) [inline src/main/java/employees/Employee.java]: abstract

File: src/main/java/employees/EmployeeWithFixedSalary.java [hunk 13]
```diff
@@ -0,0 +1,27 @@
+package employees;
+
+import departmentsAndCompany.Company;
+import departmentsAndCompany.Department;
+import salary.ISalary;
+
+public class EmployeeWithFixedSalary extends Employee implements ISalary {
```
Comment (toleksyn) [inline src/main/java/employees/EmployeeWithFixedSalary.java]: FixedSalaryEmployee

File: src/main/java/employees/EmployeeWithFixedSalary.java [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+        super(firstname, lastname, id, company, department);
+        this.monthSalary = monthSalary;
+    }
+
+    @Override
+    public double getAverageSalary() {
+        return monthSalary;
+    }
+
+    public String toString() {
```
Comment (toleksyn) [inline src/main/java/employees/EmployeeWithFixedSalary.java]: move to base class

File: src/main/java/employees/EmployeeWithSalaryPerHour.java [hunk 15]
```diff
@@ -0,0 +1,37 @@
+package employees;
+
+import departmentsAndCompany.Company;
+import departmentsAndCompany.Department;
+import salary.ISalary;
+
+public class EmployeeWithSalaryPerHour extends Employee implements ISalary {
```
Comment (toleksyn) [inline src/main/java/employees/EmployeeWithSalaryPerHour.java]: HourlyPayEmployee

File: src/main/java/employees/EmployeeWithSalaryPerHour.java [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+    public double getWorkedHoursPerMonth() {
+        return workedHoursPerMonth;
+    }
+
+    @Override
+    public double getAverageSalary() {
+        return getRatePerHour() * getWorkedHoursPerMonth();
+    }
+
+    public String toString() {
```
Comment (toleksyn) [inline src/main/java/employees/EmployeeWithSalaryPerHour.java]: move to base class

File: src/main/java/salary/ISalary.java [hunk 17]
```diff
@@ -0,0 +1,7 @@
+package salary;
+
+import employees.Employee;
+
+public interface ISalary {
```
Comment (toleksyn) [inline src/main/java/salary/ISalary.java]: ISalaryEmployee

File: src/main/java/salary/ISalary.java [hunk 18]
```diff
@@ -0,0 +1,7 @@
...
+    double getAverageSalary();
+}
```
Comment (toleksyn) [inline src/main/java/salary/ISalary.java]: .gitignore

File: src/main/java/SalaryTaskDemo.java (hunk 1 above)
Comment (sovchenko) [reply src/main/java/SalaryTaskDemo.java]: done

File: src/main/java/SalaryTaskDemo.java (hunk 2 above)
Comment (sovchenko) [reply src/main/java/SalaryTaskDemo.java]: done

File: src/main/java/SalaryTaskDemo.java (hunk 3 above)
Comment (sovchenko) [reply src/main/java/SalaryTaskDemo.java]: done

File: src/main/java/departmentsAndCompany/Company.java (hunk 7 above)
Comment (sovchenko) [reply src/main/java/departmentsAndCompany/Company.java]: done

File: src/main/java/departmentsAndCompany/Company.java (hunk 4 above)
Comment (sovchenko) [reply src/main/java/departmentsAndCompany/Company.java]: done

File: src/main/java/departmentsAndCompany/Company.java (hunk 5 above)
Comment (sovchenko) [reply src/main/java/departmentsAndCompany/Company.java]: done

File: src/main/java/departmentsAndCompany/Department.java (hunk 6 above)
Comment (sovchenko) [reply src/main/java/departmentsAndCompany/Department.java]: done

File: src/main/java/departmentsAndCompany/MarketingDepartment.java (hunk 10 above)
Comment (sovchenko) [reply src/main/java/departmentsAndCompany/MarketingDepartment.java]: done

File: src/main/java/departmentsAndCompany/SalesDepartment.java (hunk 11 above)
Comment (sovchenko) [reply src/main/java/departmentsAndCompany/SalesDepartment.java]: done

File: src/main/java/employees/Employee.java (hunk 12 above)
Comment (sovchenko) [reply src/main/java/employees/Employee.java]: done

Comment (sovchenko) [reply src/main/java/employees/Employee.java]: done

File: src/main/java/employees/EmployeeWithFixedSalary.java (hunk 13 above)
Comment (sovchenko) [reply src/main/java/employees/EmployeeWithFixedSalary.java]: done

File: src/main/java/employees/EmployeeWithFixedSalary.java (hunk 14 above)
Comment (sovchenko) [reply src/main/java/employees/EmployeeWithFixedSalary.java]: done

File: src/main/java/employees/EmployeeWithSalaryPerHour.java (hunk 15 above)
Comment (sovchenko) [reply src/main/java/employees/EmployeeWithSalaryPerHour.java]: done

File: src/main/java/employees/EmployeeWithSalaryPerHour.java (hunk 16 above)
Comment (sovchenko) [reply src/main/java/employees/EmployeeWithSalaryPerHour.java]: done

File: src/main/java/salary/ISalary.java (hunk 17 above)
Comment (sovchenko) [reply src/main/java/salary/ISalary.java]: done

**Pull request closed without merging**